### PR TITLE
implemented some caller id functionality

### PR DESCRIPTION
**Pull request by weissgilad**
created check_number filter that talks to call2me.co.il and checks if it knows anything

### CODE REVIEW

File: echo_filter.py [hunk 1]
```diff
@@ -12,6 +13,37 @@
...
+        return False
+    number_id_page.close()
+
+    # find result for requested number
+    result = tree.xpath("//div[contains(@class, 'resultData')]")
+    if len(result) < 1:
+        if "BLOCKED" in str(number_id_page.content):
+            update.message.reply_text("חפרת")
+            # keep DOS from denying us as well, so return False but still let users know that they are annoying
+            return False
```
Comment (weissgilad) [inline echo_filter.py]: redudnant

- [x] remove

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+lxml
```
Comment (neriat) [inline requirements.txt]: pin version.
check what you've install on your environment (pip freeze) and pin it
i.e. your version is 1.2.3, write:
lxml~=1.2.3

**Review by neriat — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (weissgilad) [reply requirements.txt]: - [x] pin version

Comment (weissgilad) [reply requirements.txt]: what is ~=?
shouldn't it be >= or are we not sure if a future update could break it?

Comment (neriat) [reply requirements.txt]: update minor and patch versions but not major.

**Review by neriat — APPROVED** [review]

**Pull request merged**